app/src/main/java/org/mozilla/fenix/library/history/viewholders/HistoryListItemViewHolder.kt

```diff
@@ -97,12 +101,30 @@ class HistoryListItemViewHolder(
         this.item = item
     }
 
-    private fun toggleHeader(headerText: String?) {
+    private fun toggleHeader(
+        historyItemTimeGroup: HistoryItemTimeGroup?,
+        pagedHistoryProvider: PagedHistoryProvider
```

Comment:
A bit odd that we have to bring the provider all the way down to the item in order to know if we need to change the button binding.

Reply:
Agreed. I'll make sure I remove this for the updated implementation.